src/processes/helper/userIdsToProcess.js

```diff
@@ -23,87 +42,88 @@ const findF2UserIds = async (userId) => {
   const f2UserIds = await UserFollowUser.findAll({
     where: {
       [Op.and]: [
-        { user_id_follower: { [Op.in]:followedUserIds} },
-        { user_id_followed: { [Op.notIn]: excludeUserIds } }
+        {user_id_follower: {[Op.in]: followedUserIds}},
+        {user_id_followed: {[Op.notIn]: excludeUserIds}}
       ]
     },
     limit: 1000,
     attributes: ['user_id_followed'],
-    raw : true
-  }).then( users => users.map(user => user.user_id_followed));
+    raw: true
+  }).then((users) => users.map((user) => user.user_id_followed));
 
-  return f2UserIds
-}
-
-const findBlockedUserIds = async (userId) => {
-  const blockedUserIds = await UserBlockedUser.findAll({
-    where: {
-      user_id_blocker: userId,
-    },
-    limit: 1000,
-    attributes: ['user_id_blocked'],
-    raw : true
-  }).then( users => users.map(user => user.user_id_blocked));
-
-  return blockedUserIds
-}
+  return f2UserIds;
+};
 
 const findRelatedUserIds = async (userId) => {
   const [followedByUserId, f2UserIds, blockedUserIds] = await Promise.all([
     findFollowingUserIds(userId),
     findF2UserIds(userId),
-    findBlockedUserIds(userId),
+    findBlockedUserIds(userId)
   ]);
 
-  let listUser = [...followedByUserId, ...f2UserIds, ...blockedUserIds];
-  let listUniqueUser = [...new Set(listUser)]
-  return listUniqueUser
-}
+  const listUser = [...followedByUserId, ...f2UserIds, ...blockedUserIds];
+  const listUniqueUser = [...new Set(listUser)];
+  return listUniqueUser;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `findRelatedUserIds` is combining the results of three different queries into one array and then removing duplicates. This could be inefficient if there are many users. Consider optimizing this by using SQL's UNION operator to combine these queries into one, which would automatically remove duplicates.



<!-- This is an auto-generated comment by OSS CodeRabbit -->